studio.assetmanager.ams.yaml

```diff
@@ -0,0 +1,77 @@
+app-id: studio.assetmanager.ams
+runtime: org.freedesktop.Platform
+runtime-version: '24.08'
+sdk: org.freedesktop.Sdk
+base: org.electronjs.Electron2.BaseApp
+base-version: '24.08'
+command: ams
+finish-args:
+  # X11 performance
+  - --share=ipc
+  # Access to network
+  - --share=network
+  # Access to X11
+  - --socket=x11
+  # Access to wayland
+  - --socket=wayland
+  # Audio Access
+  - --socket=pulseaudio
+  # Access to all files
+  - --filesystem=host
+  # Access to OpenGL
+  - --device=dri
+  # Use same mouse cursors as host
+  - --env=XCURSOR_PATH=/run/host/user-share/icons:/run/host/share/icons
+  - --talk-name=org.freedesktop.secrets
+
+rename-desktop-file: asset-manager-studio.desktop
+rename-icon: asset-manager-studio
+modules:
+  - name: ams-desktop
+    buildsystem: simple
+    build-commands:
+      # Extract .deb & cleanup
+      - ar -x asset-manager-studio*.deb
+      - rm -f asset-manager-studio*.deb
+      - tar -xf data.tar.xz
+      - rm -f control.tar.gz data.tar.xz debian-binary
+
+      # Copy application to Flatpack destination, and then clean it up
+      - cp --preserve --recursive opt/* usr/* $FLATPAK_DEST
+      - rm -rf opt usr
+
+      # Install our two data files
+      - install ams.sh $FLATPAK_DEST/bin/ams
+      - install -Dm0644 "${FLATPAK_ID}.metainfo.xml" "${FLATPAK_DEST}/share/metainfo/${FLATPAK_ID}.metainfo.xml"
+
+      # update Desktop file
+      - desktop-file-edit --set-key=Exec --set-value="ams %U" $FLATPAK_DEST/share/applications/asset-manager-studio.desktop
+      - desktop-file-edit --set-key=Icon --set-value="${FLATPAK_ID}" "${FLATPAK_DEST}/share/applications/asset-manager-studio.desktop"
+      - desktop-file-edit --set-key=X-Flatpak-RenamedFrom --set-value='asset-manager-studio.desktop;' "${FLATPAK_DEST}/share/applications/asset-manager-studio.desktop"
+
+      # required to patch the app.asar file of an extracted application
+      - patch-desktop-filename "${FLATPAK_DEST}/asset-manager-studio/resources/app.asar"
+
+      # To allow separate locales
+      # https://searchfox.org/mozilla-central/rev/8a4f55bc09ffc5c25dcb4586c51ae4a9fee77b4c/taskcluster/docker/firefox-flatpak/runme.sh#131-133
+      - |
+        for lang in ${FLATPAK_DEST}/asset-manager-studio/locales/*.pak
+        do
+          locale="$(basename -s .pak $lang)"
+          install -Dm644 -t "${FLATPAK_DEST}/share/runtime/locale/${locale%%-*}/" "$lang"
+          ln -sf "${FLATPAK_DEST}/share/runtime/locale/${locale%%-*}/$(basename $lang)" "${FLATPAK_DEST}/asset-manager-studio/locales/$(basename $lang)"
+        done
+    sources:
+      - type: file
+        only-arches:
+          - x86_64
+        url: https://assetmanager.studio/dl/0.0.27/asset-manager-studio_0.0.27_amd64.deb
```

Comment:
why not building from source?

Reply:
Well in this case the .so file that is built using an older version of the OS GLIBC so that it works on virtually all linux machines.   Not sure if this impacts Flatpak, but it does impact the AppImage/DEB versions.   ;-)   

Second It is far simpler to extract out everything from a  working .deb file like many of the other Electron apps do, since the Electron-builders flatpak support is extremely poor.

Reply:
so basically you just do like all the others. and don't want to provide an aarch64 which you'd get building from source, since this seems to be the only reason there is no aarch64 build here.

Reply:
1. Electron-builder's support for Flatpak can only generate a standalone .flatpak not one that uses the proper image files.    At this point switching build systems would be very complex as I have multiple custom rules so that my CI/CD machines can build both Linux and Windows and sign the windows apps at the same time...

2. The .so file is **critically** important that it is linked (at least for the .deb/appimage) to the oldest GLIBC I can support, because Ubuntu 18, 20 (and many other distro's) use an older GLibC.   If I let the "default" build occur  I get an app that crashes on a decent chunk of linux machines because of GLibC issues.     I honestly don't know if this affects flatpak's...